core/src/subgraph/runner.rs

```diff
@@ -657,11 +717,6 @@ where
             );
             block_state.persist_data_source(data_source.as_stored_dynamic_data_source());
         }
-
-        // Merge filters from data sources into the block stream builder
-        self.ctx
-            .filter
-            .extend(data_sources.iter().filter_map(|ds| ds.as_onchain()));
```

Comment:
We dont need to Merge the filters anymore since we rebuild the filters each time the block stream is restarted. And block stream is restarted everytime an onchain data source is created